filter.php

```diff
@@ -35,7 +35,8 @@ public function filter($text, array $options = []) {
             return $text;
         }
         $language = current_language();
-        if ($CFG->lang == $language) {
+
+        if (empty(get_config('filter_fulltranslate', 'translatewhensitedefault')) && $CFG->lang == $language) {
```

Comment:
1. Can the order of conditions be reversed so that default language and current language are compared first, and only if they are the same, the configuration is checked. This is just a micro-optimalisation as text comparison is generally cheaper that checking config.
2. When comparing texts such as `$CFG->lang` and `$language` it is better and safer to use identity operator in PHP. Can we please change it to `===`.

Reply:
All done in the next commit. Thanks for the feedback.